javascript/index.js

```diff
@@ -28,13 +28,119 @@
 
 
 // Iteration 1 - using callbacks
-// ...
+
+getInstruction("mashedPotatoes", 0, (step1) => {
+  document.querySelector("#mashedPotatoes").innerHTML += `<li>${step1}</li>`;
+  getInstruction("mashedPotatoes", 1, (step2) => {
+    document.querySelector("#mashedPotatoes").innerHTML += `<li>${step2}</li>`;
+    getInstruction("mashedPotatoes", 2, (step3) => {
+      document.querySelector("#mashedPotatoes").innerHTML += `<li>${step3}</li>`;
+      getInstruction("mashedPotatoes", 3, (step4) => {
+        document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`;
+        getInstruction("mashedPotatoes", 4, (step5) => {
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>${step5}</li>`;
+          document.querySelector("#mashedPotatoes").innerHTML += `<li> Mashed Potatoes are ready</li>`
+          document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
+        }, (error) => console.log(error));
+      }, (error) => console.log(error));
+    }, (error) => console.log(error));
+  }, (error) => console.log(error));
+}, (error) => console.log(error));
+
+
 
 // Iteration 2 - using promises
-// ...
+
+obtainInstruction('steak', 0)
+  .then(response=> {
+    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
+  }).then(function() {
+    return obtainInstruction('steak', 1)
+  })
+  .then(response=> {
+    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
+  })
+  .then(function() {
+    return obtainInstruction('steak', 2)
+  })
+  .then(response=> {
+    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
+  })
+  .then(function() {
+    return obtainInstruction('steak', 3)
+  })
+  .then(response=> {
+    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
+  })
+  .then(function() {
+    return obtainInstruction('steak', 4)
+  })
+  .then(response=> {
+    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
+  })
+  .then(function() {
+    return obtainInstruction('steak', 5)
+  })
+  .then(response=> {
+    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
+  })
+  .then(function() {
+    return obtainInstruction('steak', 6)
+  })
+  .then(response=> {
+    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
+  })
+  .then(function() {
+    return obtainInstruction('steak', 7)
+  })
+  .then(response=> {
+    document.querySelector("#steak").innerHTML += `<li>${response}</li>`;
+  })
+  .then(function(){
+    document.querySelector("#steak").innerHTML += `<li>Steak is ready</li>`
+  })
+  .finally(function(){
+    document.querySelector("#steakImg").removeAttribute("hidden")
+  })
```

Comment:
Same note as above, tasks to update DOM are synchronous, so you could list them in one promise